src/compiler/crystal/tools/print_hierarchy.cr

```diff
@@ -3,8 +3,13 @@ require "colorize"
 require "../syntax/ast"
 
 module Crystal
-  def self.print_hierarchy(program, exp)
-    HierarchyPrinter.new(program, exp).execute
+  def self.print_hierarchy(program, exp, format)
+    case format
+    when "json"
+      JSONHierarchyPrinter.new(program, exp).execute
+    else
```

Comment:
The else branch should rather abort and tell the user about the invalid parameter argument.